main.go

```diff
@@ -12,18 +12,28 @@ func main() {
 	interrupt := make(chan os.Signal, 1)
 	signal.Notify(interrupt, os.Interrupt)
 
+	// it is better that each entity will do its own setup. Less parameters when you call entities.NewXXXXX
+	// it will also be clearer which values are needed for each entity.
+	// will be great if you have a different setup for each entity (.env and config.json can be common).
 	app := setup()
 
 	shutdownChan := make(chan struct{})
+	// swapIntentOperationSubmitChan used by user to submit swapIntentOperations to bundler
 	swapIntentOperationSubmitChan := make(chan *entities.SwapIntentOperation)
+	// swapIntentOperationBroadcastChan used by bundler to announce swapIntentOperations to solver
 	swapIntentOperationBroadcastChan := make(chan *entities.SwapIntentOperation)
+	// solverOperationSubmitChan used by solver to submit solverOperation back to bundler
 	solverOperationSubmitChan := make(chan *Atlas.SolverOperation)
+	// governanceSignatureChan - ??????
 	governanceSignatureChan := make(chan []byte)
 
+	// Q: what is the governance entity? It is not clear why governance is needed and when?
+	// Q: is this an online component (service) that the dapp has to provide?
 	// Launch the governance goroutine
 	governance := entities.NewGovernance(app.PrivateKeys["governance"], governanceSignatureChan, shutdownChan)
```

Comment:
The governance is the entity signing the dApp operation, which is a required component for the Atlas metacall function.

In this example we've made the governance a separate entity, as if the backend and the governance were different parties. In that case, the governance would have to provide a service for signing operations requested by the backend. For your use case, the governance would probably be integrated into your backend, since you would be both those entities.

Reply:
OK. Got it.